tasklist/client/src/modules/queries/useProcessDefinition.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {useQuery} from '@tanstack/react-query';
+import {api} from 'modules/api';
+import {RequestError, request} from 'modules/request';
+import {Process} from 'modules/types';
+
+const HTTP_STATUS_FORBIDDEN = 403;
+const HTTP_STATUS_NOT_FOUND = 404;
+
+type Data = Process;
+
+function useProcessDefinition(
+  processDefinitionId: string | undefined | null,
+  options?: {enabled?: boolean},
+) {
+  return useQuery<Data, RequestError>({
+    queryKey: ['processDefinition', processDefinitionId],
+    queryFn: async () => {
+      const {response, error} = await request(
+        api.getProcess({processDefinitionId: processDefinitionId!}),
+      );
+
+      if (error !== null && error.variant === 'failed-response') {
+        const {status} = error.response;
+        if (
+          status === HTTP_STATUS_FORBIDDEN ||
+          status === HTTP_STATUS_NOT_FOUND
+        ) {
+          return null;
+        }
+      }
```

Comment:
The behaviour for a 404 Not Found error is different to a 500 Internal Server Error. I think we should explicitly handle business cases (not found and forbidden) since it makes it simpler to whoever consumes this hook.

Reply:
> The behaviour for a 404 Not Found error is different to a 500 Internal Server Error. I think we should explicitly handle business cases (not found and forbidden) since it makes it simpler to whoever consumes this hook.

but this is basically goes against react-query conventions of throwing errors and it makes harder for the hook to be reused if want to have a different behavior. It would be impossible to build a 404 page for this for example.

I believe this logic should exist in the view layer